portfolio/src/main/java/com/google/sps/servlets/CalendarServlet.java

```diff
@@ -0,0 +1,33 @@
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import com.google.sps.data.CalendarEvent;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles calendar event creation. */
```

Comment:
From what I can tell, currently the client can create calendar events on the fly through the server. I think it can later cause an issue where it's difficult to reconcile the client state vs server state.

I would recommend that you maintain the states on the client, until the schedule computation happens. This is also worthwhile to cover in the design doc. What kind of APIs will the client invoke to the server?

Consider the following approach, for example.
1. Suppose the user created a calendar event. At this point, client can just hold onto the event, and not tell the sever yet.
2. Similarly, client can create tasks that are kept track on the client side.
3. Now when the user tries to compute the schedule, client can call some API that takes the tasks and events, and returns the computed schedule.

In other words, while keeping track of these intermediate states on the server can be a good idea, it brings in complexity. And at this stage you may want to avoid keeping track of states on both the client and the server. Consider making the server operations state-less at this time.